src/Service.Tests/OpenApiDocumentor/PathValidation.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.DataApiBuilder.Config.ObjectModel;
+using Microsoft.OpenApi.Models;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Azure.DataApiBuilder.Service.Tests.OpenApiIntegration
+{
+    /// <summary>
+    /// Integration tests validating correct keys are created for OpenApiDocument.Paths
+    /// which represent the path used to access an entity in DAB's REST API endpoint.
+    /// </summary>
+    [TestCategory(TestCategory.MSSQL)]
+    [TestClass]
+    public class PathValidation
+    {
+        private const string CUSTOM_CONFIG = "path-config.MsSql.json";
+        private const string MSSQL_ENVIRONMENT = TestCategory.MSSQL;
+
+        // Error messages
+        private const string PATH_GENERATION_ERROR = "Unexpected path value for entity in OpenAPI description document: ";
+
+        /// <summary>
+        /// Validates that the OpenApiDocument object's Paths property for an entity is generated
+        /// with the entity's explicitly configured REST path, if set. Otherwise, the top level
+        /// entity name is used.
+        /// </summary>
+        /// <param name="entityName">Top level entity name defined in runtime config.</param>
+        /// <param name="restPath">Entity's configured REST path.</param>
+        /// <param name="expectedOpenApiPath">Expected path generated for OpenApiDocument.Paths</param>
+        [DataRow("entity", "/customEntityPath", "/customEntityPath", DisplayName = "Entity REST path has leading slash - REST path override used.")]
+        [DataRow("entity", "//customEntityPath", "/customEntityPath", DisplayName = "Entity REST path has leading slashes - REST path override used.")]
```

Comment:
If we trim only one slash, and OpenApiDocumentor somehow still adds a slash before the expected path as per all the other DataRows, shouldn't the expected path here be `//customEntityPath`

Reply:
Added clarifying comments. The OpenApi path value created takes the form `"/{entityRestPath}"` where `{entityRestPath}` is stripped of all leading slashes when openapidocumentor evaluates whether to use the entity name or configured rest path as the value of rest path to use.